src/test/java/org/prebid/server/bidder/openx/OpenxBidderTest.java

```diff
@@ -0,0 +1,467 @@
+                .imp(asList(
+                        Imp.builder().id("impId1").xNative(Native.builder().build()).build(),
+                        Imp.builder().id("impId2").xNative(Native.builder().build()).build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = openxBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors()).hasSize(2)
+                .extracting(BidderError::getMessage)
+                .containsExactly(
+                        "OpenX only supports banner and video imps. Ignoring imp id=impId1",
+                        "OpenX only supports banner and video imps. Ignoring imp id=impId2");
+    }
+
+    @Test
+    public void makeHttpRequestShouldReturnResultWithErrorWhenImpExtOmitted() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList((
+                        Imp.builder().id("impId1").banner(Banner.builder().build()).build()),
+                        Imp.builder().id("impId2").video(Video.builder().build()).build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = openxBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors()).hasSize(2)
+                .extracting(BidderError::getMessage)
+                .containsExactly("openx parameters section is missing", "openx parameters section is missing");
+    }
+
+    @Test
+    public void makeHttpRequestShouldReturnResultWithErrorWhenImpExPrebidtMalformed() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList((
+                        Imp.builder()
+                                .id("impId1")
+                                .banner(Banner.builder().build())
+                                .ext(mapper.valueToTree(new HashMap<>())).build()),
+                        Imp.builder()
+                                .id("impId2")
+                                .video(Video.builder().build())
+                                .ext(mapper.valueToTree(new HashMap<>())).build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = openxBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors()).hasSize(2)
+                .extracting(BidderError::getMessage)
+                .containsExactly("openx parameters section is missing", "openx parameters section is missing");
+    }
+
+    @Test
+    public void makeHttpRequestShouldReturnResultWithErrorWhenImpExtOpenxEmpty() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList(
```

Comment:
Replace asList() call with singletonList() at lines : 158, 180, 294, 316, 394 and 437.